File: trimester-2/10_mouse_input_review_peer/program_files/cpp/7mouseposvector.cpp
#include "splashkit.h"

int main()
{
    open_window("Mouse Location", 800, 600);

    while (!window_close_requested("Mouse Location"))
    {
        // process user input events
        process_events();

        // draw line from mouse position vector
        clear_screen(COLOR_WHITE);
        draw_line(COLOR_BLUE, line_from(mouse_position_vector()));
        refresh_screen(60);
    }

    close_all_windows();

    return 0;
}

File: peer-reviews/10_mouse_input_review_peer/program_files/python/7mouseposvector.py
from splashkit import *

def main():

    window = open_window("Mouse Location", 800, 600)

    while not window_close_requested(window):
        # Process user input events
        process_events()

        # Clear the screen with white color
        clear_screen(color_white())

        # Get the current mouse position
        mouse_pos = mouse_position_vector()

        # Draw a line from the top-left corner to the current mouse position
        draw_line(color_blue(), 0, 0, mouse_pos.x, mouse_pos.y)


        refresh_window(window)


    close_window(window)

if __name__ == "__main__":
    main()